karaf/deploy-and-run.sh

```diff
@@ -1,23 +1,41 @@
 #!/bin/sh
 
-DIR="/deployments"
+DIR="/deployments/"
 
-echo "Checking for *.tar.gz in $DIR"
-# there should be only one *.tar.gz in ${DIR}
-NUM_ARCHIVE_FILES=`ls -1 ${DIR} | grep "^.*.tar.gz$" | wc -l`
-if [ $NUM_ARCHIVE_FILES -ne 1 ]; then
-  echo "Missing or more than one assembly archive file *.tar.gz in ${DIR}"
-  exit 1
-fi
-KARAF_ASSEMBLY_ARCHIVE=`ls -1 ${DIR}/*.tar.gz`
+echo "Checking for *.zip in $DIR"
+
+if [ `ls -1 ${DIR} | grep "^.*.zip$" | wc -l` -eq 1 ]; then
+  #
+  # zip
+  # 
+  KARAF_ASSEMBLY_ARCHIVE=`ls -1 ${DIR}/*.zip`
+  echo "Found $KARAF_ASSEMBLY_ARCHIVE in ${DIR}"
+  
+  # extract custom assembly to DEPLOY_DIR
+  unzip "$KARAF_ASSEMBLY_ARCHIVE" -d ${DIR}
```

Comment:
Unzip is not available in every base image by default (luckily for this base image it is). Maybe we could use `jar` here, too (as for java based applications it should be always available, even when we change the base image) ?